packages/tldraw/src/lib/ui/components/Toolbar/DefaultImageToolbarContent.tsx

```diff
@@ -0,0 +1,237 @@
+		: 0
+	useEffect(() => {
+		if (isManipulating && maxZoom === undefined) {
+			setMaxZoom(Math.max(zoom, 1 - 1 / MAX_ZOOM))
+		} else if (!isManipulating) {
+			setMaxZoom(undefined)
+		}
+	}, [isManipulating, zoom, maxZoom])
+
+	const insertMedia = useInsertMedia({ shapeIdToReplace: imageShape?.id })
+
+	const onHistoryMark = useCallback((id: string) => editor.markHistoryStoppingPoint(id), [editor])
+
+	const handleZoomChange = useCallback(
+		(value: number) => {
+			editor.setCurrentTool('select.crop.idle')
+			const zoom = value / 100
+			const change = getCroppedImageDataWhenZooming(zoom, imageShape, maxZoom)
+
+			editor.updateShape({
+				id: imageShape.id,
+				type: imageShape.type,
+				x: change.x,
+				y: change.y,
+				props: {
+					w: change.w,
+					h: change.h,
+					crop: change.crop,
+				},
+			} as TLShapePartial)
+
+			trackEvent('set-style', { source: 'image-menu', id: 'zoom', value })
+		},
+		[editor, trackEvent, imageShape, maxZoom]
+	)
+
+	const handleAspectRatioChange = (aspectRatio: ASPECT_RATIO_OPTION) => {
+		editor.setCurrentTool('select.crop.idle')
+		const change = getCroppedImageDataForAspectRatio(aspectRatio, imageShape)
+		if (!change) return
+
+		editor.markHistoryStoppingPoint('aspect ratio')
+		editor.updateShape({
+			id: imageShape.id,
+			type: imageShape.type,
+			x: change.x,
+			y: change.y,
+			props: {
+				crop: change.crop,
+				w: change.w,
+				h: change.h,
+			},
+		} as TLShapePartial)
+	}
+
+	const shapeAspectRatio = imageShape.props.w / imageShape.props.h
+	const isOriginalCrop = !crop || isEqual(crop, getDefaultCrop())
+	const displayValue = crop && maxZoom ? (100 / maxZoom) * zoom : 0
+
+	const croppingTools = (
+		<>
+			<TldrawUiSlider
+				value={displayValue}
+				label="tool.image-zoom"
+				onValueChange={handleZoomChange}
+				onHistoryMark={onHistoryMark}
+				min={0}
+				steps={100}
+				data-testid="tool.image-zoom"
+				title={msg('tool.image-zoom')}
+			/>
+			<TldrawUiDropdownMenuRoot id="image-toolbar-aspect-ratio">
+				<TldrawUiDropdownMenuTrigger>
+					<TldrawUiButton title={msg('tool.aspect-ratio')} type="icon">
+						<TldrawUiButtonIcon icon="corners" />
+					</TldrawUiButton>
+				</TldrawUiDropdownMenuTrigger>
+				<TldrawUiDropdownMenuContent side="top" align="center">
+					{ASPECT_RATIO_OPTIONS.map((aspectRatio) => (
+						<TldrawUiDropdownMenuCheckboxItem
+							key={aspectRatio}
+							onSelect={() => handleAspectRatioChange(aspectRatio as ASPECT_RATIO_OPTION)}
+							checked={
+								aspectRatio === 'circle' && crop
+									? crop.isCircle
+									: aspectRatio === 'original'
+										? isOriginalCrop
+										: aspectRatio === 'square'
+											? !crop?.isCircle &&
+												approximately(shapeAspectRatio, ASPECT_RATIO_TO_VALUE[aspectRatio], 0.1)
+											: !isOriginalCrop &&
+												approximately(shapeAspectRatio, ASPECT_RATIO_TO_VALUE[aspectRatio], 0.01)
```

Comment:
nit: i usually don't mind nested ternaries but this one is pretty intense. worth pulling out to a couple variables so we can see what's going on?